apps/api/src/app/integrations/usecases/update-integration/update-integration.usecase.ts

```diff
@@ -0,0 +1,49 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository } from '@notifire/dal';
+import { UpdateIntegrationCommand } from './update-integration.command';
+
+@Injectable()
+export class UpdateIntegration {
+  constructor(private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: UpdateIntegrationCommand): Promise<IntegrationEntity> {
+    const existingIntegration = await this.integrationRepository.findById(command.integrationId);
```

Comment:
I would add the _applicationId to the query here, because otherwise, every user requesting every integration by id can access it. We want at least one authentication param from JWT token to. validate the userability to edit the resource.

Reply:
@djabarovgeorge How did you ended up solving this? I still see that it's not querying with the application id from jwt